### PR TITLE
Close out NASR dynamic frequencies (#199)

### PR DESCRIPTION
**Pull request by alexwitherspoon**
## Summary
- Document read-time frequency merge in `CONFIGURATION.md` (config → NASR FRQ → OurAirports; NASR preferred when both exist).
- Expand NASR FRQ parsing: primary, IC fallback, and secondary tiers; IC preference for approach/departure; pairing metadata for CTAF/UNICOM and tower/CTAF companion suppression.
- Merge layer: `clearance_delivery` alias, per-field config override, and suppress ghost companion roles when config overrides a paired NASR frequency.
- Bump `NASR_FRQ_SCHEMA_VERSION` to 3; cache payload includes `pairing` metadata written by `scripts/fetch-nasr-frq.php`.
- Add fixture-backed unit tests for international NASR-over-OurAirports precedence (`CYVR`), tier mapping edge cases, companion suppression, and Public API `formatAirportDetails` merge behavior.

Closes #199

## Context
Implementation (FRQ ingest, merge lib, dashboard + `GET /v1/airports/{id}` wiring) landed with reference-data observability (#229). This PR completes #199 with parser/merge hardening, docs, tests, and cache schema v3.

**Operator note:** deploy expects a NASR FRQ cache refresh (`scripts/fetch-nasr-frq.php` or scheduler) so pairing metadata and tier v3 mappings are populated.

## Test plan
- [x] `make test-ci`
- [x] Prod spot-check: `cache/nasr/nasr_frq.json` fresh; US airport without config `frequencies` shows merged values on dashboard + Public API

### CODE REVIEW

File: tests/Unit/PublicApiAirportTest.php [hunk 1]
```diff
@@ -9,8 +9,17 @@
...
+
 class PublicApiAirportTest extends TestCase
 {
+    use LoadsNasrFrqFixtureCacheTrait;
+
+    protected function tearDown(): void
+    {
+        $this->tearDownNasrFrqFixtureCache();
+        resetOurAirportsFrequenciesCacheMemo();
+    }
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply tests/Unit/PublicApiAirportTest.php:31]: Good catch - agreed. Fixed in 1867e9c5: tearDown now requires airport-frequencies.php before resetOurAirportsFrequenciesCacheMemo() and calls parent::tearDown().

File: tests/Unit/AirportFrequenciesTest.php [hunk 2]
```diff
@@ -150,6 +150,29 @@ public function testMergedFrequenciesUseOurAirportsWhenNasrMissing(): void
         $this->assertSame(['ctaf' => '123.2'], $merged);
     }
 
+    public function testMergedFrequenciesPreferNasrOverOurAirportsWhenBothPresent(): void
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply tests/Unit/AirportFrequenciesTest.php]: Agreed - renamed in 45c698f1 to testGetMergedAirportFrequencies_NasrAndOurAirportsBothPresent_PrefersNasr() per CODE_STYLE.md.

File: tests/Unit/PublicApiAirportTest.php [hunk 3]
```diff
@@ -9,8 +9,19 @@
...
+    use LoadsNasrFrqFixtureCacheTrait;
+
+    protected function tearDown(): void
+    {
+        $this->tearDownNasrFrqFixtureCache();
+        require_once __DIR__ . '/../../lib/airport-frequencies.php';
+        resetOurAirportsFrequenciesCacheMemo();
+        parent::tearDown();
+    }
     private static function loadFormatAirportDetails(): void
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply tests/Unit/PublicApiAirportTest.php:33]: Agreed - added the blank line in ec904256.

File: tests/Unit/PublicApiAirportTest.php [hunk 4]
```diff
@@ -224,4 +236,60 @@ public function testFormatAirportDetails_ServicesAndFrequenciesReturnObjectWhenP
         $this->assertArrayHasKey('fuel', $formatted['services']);
         $this->assertSame('100LL, Jet-A', $formatted['services']['fuel']);
     }
+
+    public function testFormatAirportDetails_MergedFrequenciesFromNasrWhenConfigMissing(): void
+    {
+        self::loadFormatAirportDetails();
+        $this->loadNasrFrqFixtureCache();
+
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply tests/Unit/PublicApiAirportTest.php:251]: Good point - fixed in f420f97b by seeding an empty OurAirports cache in setUp() for the whole test class (same pattern for AirportFrequenciesTest).

File: tests/Unit/PublicApiAirportTest.php [hunk 5]
```diff
@@ -224,4 +236,60 @@ public function testFormatAirportDetails_ServicesAndFrequenciesReturnObjectWhenP
...
+            'ground' => '121.7',
+            'atis' => '124.6',
+        ], $formatted['frequencies']);
+    }
+
+    public function testFormatAirportDetails_ConfigFrequencyOverridesNasrPerField(): void
+    {
+        self::loadFormatAirportDetails();
+        $this->loadNasrFrqFixtureCache();
+
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply tests/Unit/PublicApiAirportTest.php:290]: Addressed in the same commit (f420f97b) via setUp() so both NASR-only tests stay hermetic.

File: tests/Unit/PublicApiAirportTest.php [hunk 6]
```diff
@@ -224,4 +243,60 @@ public function testFormatAirportDetails_ServicesAndFrequenciesReturnObjectWhenP
...
+        $formatted = formatAirportDetails('69v', [
+            'faa' => '69V',
+            'name' => 'Huntington Muni',
+            'frequencies' => [
+                'ctaf' => '123.0',
+            ],
+        ]);
+
+        $this->assertSame('123', $formatted['frequencies']['ctaf']);
+        $this->assertSame('122.8', $formatted['frequencies']['unicom']);
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply tests/Unit/PublicApiAirportTest.php]: Good catch - agreed. Updated the assertion in 09e6993f so ctaf-only config override no longer expects a ghost unicom after companion-role suppression.

File: lib/nasr/frequencies-parse.php [hunk 7]
```diff
@@ -73,37 +255,26 @@ function nasrParseFrqCsvFile(string $csvPath): array
...
+    foreach (nasrIterateCsvFile($csvPath) as $row) {
+        nasrApplyFrqRowToAirports($airports, $approachDepartureSources, $row, NASR_FREQ_MAP_TIER_IC_FALLBACK);
+    }
 
-        // First VHF row per role wins (e.g. skip secondary LCL/S when LCL/P already set).
-        if (!isset($airports[$arptId][$role])) {
-            $airports[$arptId][$role] = $mhz;
-        }
+    foreach (nasrIterateCsvFile($csvPath) as $row) {
+        nasrApplyFrqRowToAirports($airports, $approachDepartureSources, $row, NASR_FREQ_MAP_TIER_SECONDARY);
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply lib/nasr/frequencies-parse.php]: Agreed - refactored in 72294eff: one CSV pass collects per-role candidates with tier/order, then nasrResolveFrqRoleCandidates() picks the winning MHz without reopening the file.

File: scripts/fetch-nasr-frq.php [hunk 8]
```diff
@@ -187,7 +187,7 @@ function fetchNasrFrqIfNeeded(bool $force = false): bool
             'frq_last_discovery_at' => gmdate('c'),
         ];
 
-        $saved = saveNasrFrqCache($parsed['airports'], $meta);
+        $saved = saveNasrFrqCache($parsed['airports'], $meta, $parsed['pairing'] ?? []);
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply scripts/fetch-nasr-frq.php:190]: Fair point - updated the PR description to call out parser/merge changes, schema v3, pairing metadata in the FRQ cache, and the operator refresh note.

File: lib/nasr/frequencies-parse.php [hunk 9]
```diff
@@ -60,50 +106,210 @@ function nasrFormatFrequencyMhz($mhz): ?string
 }
 
 /**
- * Parse NASR FRQ.csv into airports keyed by SERVICED_FACILITY (ARPT_ID).
+ * @param array<string, array{primary: ?string, ic: ?string}> $approachDepartureSources
+ */
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply lib/nasr/frequencies-parse.php:110]: Good catch - agreed. Fixed in a2d41d19: the annotation now matches the ARPT_ID -> role -> {primary, ic} shape used by nasrApplyApproachDepartureInitialContactPreference().

File: lib/nasr/frequencies-parse.php [hunk 10]
```diff
@@ -60,50 +106,210 @@ function nasrFormatFrequencyMhz($mhz): ?string
...
+    }
+}
+
+/**
+ * Apply one FRQ.csv row to per-airport role candidates.
  *
- * @param string $csvPath Path to FRQ.csv
- * @return array{airports: array<string, array<string, string>>, effective_date: ?string}
+ * @param array<string, array<string, list<array{mhz: string, tier: int, order: int}>>> $roleCandidates
+ * @param array<string, array{primary: ?string, ic: ?string}> $approachDepartureSources
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply lib/nasr/frequencies-parse.php]: Same fix in a2d41d19 - updated the nasrCollectFrqRowCandidates() param doc to the nested ARPT_ID/role contract.

File: lib/nasr/frequencies-parse.php [hunk 11]
```diff
@@ -60,50 +106,210 @@ function nasrFormatFrequencyMhz($mhz): ?string
...
+
+    if (!isset($approachDepartureSources[$arptId][$role])) {
+        $approachDepartureSources[$arptId][$role] = ['primary' => null, 'ic' => null];
+    }
+
+    if ($tier === NASR_FREQ_MAP_TIER_PRIMARY) {
+        $approachDepartureSources[$arptId][$role]['primary'] = $mhz;
+    } elseif ($tier === NASR_FREQ_MAP_TIER_IC_FALLBACK) {
+        $approachDepartureSources[$arptId][$role]['ic'] = $mhz;
+    }
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply lib/nasr/frequencies-parse.php:134]: Good catch - agreed. Fixed in 83543e8f: primary/IC source tracking now keeps the first MHz seen per tier so IC preference compares against the same primary row nasrResolveFrqRoleCandidates() selected.

File: docs/CONFIGURATION.md [hunk 12]
```diff
@@ -182,11 +182,15 @@ Policy detail: `docs/SAFETY_CRITICAL_CALCULATIONS.md` (Density Altitude Performa
...
+
+NASR `FREQ_USE` mappings include tower/ground (`LCL/P`, `GND/P`), CTAF/UNICOM, ATIS, approach/departure (`APCH/P`, `DEP/P`, combined `APCH/P DEP/P`), clearance delivery (`CD/P`), and automated weather rows whose `FREQ_USE` contains `ASOS` or `AWOS`. Initial Contact rows (`LCL/P IC`, `APCH/P IC`, `DEP/P IC`, `APCH/P DEP/P IC`) fill a role when no primary `/P` row exists for that role. Secondary NASR rows (`LCL/S`, `GND/S`, `APCH/S`, `DEP/S`, and `CD PRE *` clearance) fill a role only when neither primary nor IC rows exist. When both primary and Initial Contact approach/departure rows exist at different MHz values, Initial Contact is shown (sector-specific frequencies may still apply; see chart supplement).
 
 | Key | When to use |
 |-----|-------------|
-| `tower`, `ground`, `atis`, `clearance_delivery`, `approach`, `departure` | As published for controlled airports. |
+| `tower`, `ground`, `atis`, `clearance`, `approach`, `departure` | As published for controlled airports. |
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply docs/CONFIGURATION.md]: Agreed - noted in ef2f7b0a that clearance_delivery is accepted and normalized to clearance at merge time.
